tests/Unit/Helpers/NumericalAlgorithms/FiniteDifference/Exact.hpp

```diff
@@ -0,0 +1,173 @@
+#include "Domain/Structure/DirectionMap.hpp"
+#include "Evolution/DgSubcell/SliceData.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace TestHelpers::fd::reconstruction {
+/*!
+ * \brief Test that the reconstruction procedure can exactly reconstruct
+ * polynomials of degree equal to the degree of the basis.
+ *
+ * For example, for a minmod, MC, or other second-order TVD limiter, we must be
+ * able to reconstruct linear functions exactly. For a third-order WENO scheme,
+ * we must be able to reconstruct quadratic functions exactly. This is done
+ * by setting up the function
+ *
+ * \f{align}{
+ *  u=\sum_{i=1}^p x^p + y^p + z^p
+ * \f}
+ *
+ * where \f$x,y,z\f$ are compute by the logical coordinates translated by
+ * \f$4\f$ in \f$y\f$ and by \f$8\f$ in \f$z\f$. The translation is done
+ * to catch subtle bugs that may arise from indexing in the wrong dimension.
+ */
+template <size_t Dim, typename F>
+void test_reconstruction_is_exact_if_in_basis(const size_t max_degree,
+                                              const size_t points_per_dimension,
+                                              const size_t stencil_width,
+                                              const F& invoke_recons) {
+  const size_t number_of_vars = 2;  // arbitrary, 2 is "cheap but not trivial"
+
+  const Mesh<Dim> mesh{points_per_dimension, Spectral::Basis::FiniteDifference,
+                       Spectral::Quadrature::CellCentered};
+  auto log_coords = logical_coordinates(mesh);
+  // Make the logical coordinates different in each direction
+  for (size_t i = 1; i < Dim; ++i) {
+    log_coords.get(i) = log_coords.get(i) + 4.0 * i;
+  }
+
+  // Compute polynomial on cell centers in FD cluster of points
+  const auto set_polynomial = [max_degree](
+                                  const gsl::not_null<DataVector*> var1_ptr,
+                                  const gsl::not_null<DataVector*> var2_ptr,
+                                  const auto& logical_coords) {
+    *var1_ptr = 0.0;
+    if (number_of_vars == 2) {
+      *var2_ptr = 100.0;  // some constant offset to distinguish the var values
+    }
+    for (size_t degree = 1; degree <= max_degree; ++degree) {
+      for (size_t i = 0; i < Dim; ++i) {
+        *var1_ptr += pow(logical_coords.get(i), degree);
+        if (number_of_vars == 2) {
+          *var2_ptr += pow(logical_coords.get(i), degree);
+        }
+      }
+    }
+  };
+  std::vector<double> volume_vars(mesh.number_of_grid_points() * number_of_vars,
+                                  0.0);
+  DataVector var1(volume_vars.data(), mesh.number_of_grid_points());
+  DataVector var2(volume_vars.data() + mesh.number_of_grid_points(),  // NOLINT
+                  mesh.number_of_grid_points());
+  set_polynomial(&var1, &var2, log_coords);
+
+  // Compute polynomial on cell centers in neighbor data.
```

Comment:
I don't understand this sentence...
- should "on cell centers" -> "at cell centers" (I expect the preposition "at" when describing a location)
- what is "in neighbor data"? (I have no idea what the preposition "in" means w.r.t. data)